### PR TITLE
Add Event Hook integration

### PR DESCRIPTION
**Pull request by yordis**
- Added the `formatter.exs` file since it causes issue running `mix format` if the file is not present and think this is a good file to have in any Elixir project.

### Notes

Probably people will want to have control over the response when something goes wrong instead of `404`.

I will strongly push back on this since from security reasons is better to give less information to external users on what actually happens.

cc: @stueccles 🚀

### CODE REVIEW

**Comment by yordis** [comment]
@stueccles do you think we could get this into the lib? I would love to remove my code 😄

**Comment by stueccles** [comment]
@yordis i'll take a look at it today :)

File: lib/okta/plug/event_hook.ex [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+      conn
+      |> authorization_token()
+      |> verify(secret_key)
+    end
+
+    defp verify(token, token), do: true
+    defp verify(_, _), do: false
+
+    defp authorization_token(conn) do
```
Comment (stueccles) [inline lib/okta/plug/event_hook.ex]: you may want to call this method something else than using the terminology `token`. makes it seem like it is an regular authorization token when in fact its just they key. maybe `request_key` ?

**Review by stueccles — APPROVED** [review]

File: test/plugs/event_hook_test.exs [hunk 2]
```diff
@@ -0,0 +1,79 @@
+defmodule Okta.Plug.EventHookTest do
+  use ExUnit.Case, async: true
+
+  alias Plug.Conn
+  alias Okta.Plug.EventHook
+
+  @authorization_token "authorization_token"
```
Comment (stueccles) [inline test/plugs/event_hook_test.exs]: might want to change naming here

**Pull request merged**